lightspeed-stack/Containerfile.lsc

```diff
@@ -0,0 +1,77 @@
+# ======================================================
+# Transient image to construct Python venv
+# ------------------------------------------------------
+FROM registry.access.redhat.com/ubi9/ubi-minimal AS builder
+
+ARG APP_ROOT=/app-root
+RUN microdnf install -y --nodocs --setopt=keepcache=0 --setopt=tsflags=nodocs \
+    python3.11 python3.11-devel python3.11-pip
+WORKDIR /app-root
+
+# UV_PYTHON_DOWNLOADS=0 : Disable Python interpreter downloads and use the system interpreter.
+ENV UV_COMPILE_BYTECODE=0 \
+    UV_LINK_MODE=copy \
+    UV_PYTHON_DOWNLOADS=0
+
+# Install uv package manager
+RUN pip3.11 install uv
+
+# Add explicit files and directories
+# (avoid accidental inclusion of local directories or env files or credentials)
+COPY lightspeed-stack/pyproject.toml lightspeed-stack/uv.lock LICENSE.md README.md ./
+
+RUN uv sync --no-install-project --no-dev
+# ======================================================
+
+# ======================================================
+# Final image without uv package manager
+# ------------------------------------------------------
+FROM quay.io/lightspeed-core/lightspeed-stack:latest
+
+USER 0
+
+ARG APP_ROOT=/app-root
+RUN microdnf install -y --nodocs --setopt=keepcache=0 --setopt=tsflags=nodocs python3.11
+WORKDIR /app-root
+
+# PYTHONDONTWRITEBYTECODE 1 : disable the generation of .pyc
+# PYTHONUNBUFFERED 1 : force the stdout and stderr streams to be unbuffered
+# PYTHONCOERCECLOCALE 0, PYTHONUTF8 1 : skip legacy locales and use UTF-8 mode
+ENV PYTHONDONTWRITEBYTECODE=1 \
+    PYTHONUNBUFFERED=1 \
+    PYTHONCOERCECLOCALE=0 \
+    PYTHONUTF8=1 \
+    PYTHONIOENCODING=UTF-8 \
+    LANG=en_US.UTF-8
+
+COPY --from=builder --chown=1001:1001 /app-root /app-root
+
+# this directory is checked by ecosystem-cert-preflight-checks task in Konflux
+COPY --from=builder /app-root/LICENSE.md /licenses/
+
+# Add executables from .venv to system PATH
+ENV PATH="/app-root/.venv/bin:$PATH"
+
+ENV LLAMA_STACK_CONFIG_DIR=/.llama/data
+
+# Data and configuration
+RUN mkdir -p /.llama/distributions/ansible-chatbot
+RUN mkdir -p /.llama/data/distributions/ansible-chatbot
+ADD ansible-chatbot-run.yaml /.llama/distributions/ansible-chatbot
+ADD providers.d /.llama/providers.d
+RUN chmod -R g+rw /.llama
+
+# lightspeed-core
+ADD embeddings_model /.llama/data/embeddings_model
```

Comment:
@manstis I am not sure about this line...

In order to run locally I understand I have to run `setup-vector-db` just before, and it works, but what will happen during the CI build, should we also run the `setup-vector-db` before the `build-lsc`?

Also, as far as I remember we're just handling the embedding model and vector database injection into the container in the k8s deployment yaml (by using a sidecontainer and mounting the volume), so what will happen if now the Containerfile depends on having this embeeded in to the image during build? no need for the k8s stuff?

Thx

Reply:
Agreed. Fixed.